### PR TITLE
[AMDGPU] Simplify renamed BUF instruction definitions. NFC.

### PR DESCRIPTION
**Pull request by jayfoad**
Use optional arguments instead of separate (multi)classes for renamed
instructions.

### CODE REVIEW

**Review by piotrAMD — APPROVED** [review]
LGTM

File: llvm/lib/Target/AMDGPU/BUFInstructions.td [hunk 1]
```diff
@@ -2895,28 +2887,25 @@ multiclass MTBUF_Real_AllAddr_gfx11_gfx12_Renamed_Impl<bits<4> op, string real_n
   defm _VBUFFER_OFFSET : VBUFFER_MTBUF_Real_gfx12<op, real_name>;
 }
 
-multiclass MTBUF_Real_AllAddr_gfx11_gfx12<bits<4> op>
- : MTBUF_Real_AllAddr_gfx11_gfx12_Renamed_Impl<op, get_BUF_ps<NAME>.Mnemonic>;
-
-
-multiclass MTBUF_Real_AllAddr_gfx11_gfx12_Renamed<bits<4> op, string real_name>
-  : MTBUF_Real_AllAddr_gfx11_gfx12_Renamed_Impl<op, real_name> {
+multiclass MTBUF_Real_AllAddr_gfx11_gfx12<bits<4> op, string real_name = !tolower(NAME)>
```
Comment (Sisyph) [inline llvm/lib/Target/AMDGPU/BUFInstructions.td:2890]:
```suggestion
multiclass MTBUF_Real_AllAddr_gfx11_gfx12<bits<4> op, string real_name = get_BUF_ps<NAME>.Mnemonic
```

It is assumed now that !tolower(NAME) == !tolower(NAME), but I think there is no need to make that assumption. Using the suggested formulation avoids it.

Comment (jayfoad) [reply llvm/lib/Target/AMDGPU/BUFInstructions.td:2890]: True - but I'm not aware of any cases where `!tolower(NAME) != get_BUF_ps<NAME>.Mnemonic`, so I'm not sure if it's a useful abstraction.

Comment (jayfoad) [reply llvm/lib/Target/AMDGPU/BUFInstructions.td:2890]: I've tried to clean this up in #84524. See what you think.

**Pull request merged**